chore(wallet)_: fee handling improvements for Status Network Sepolia (cherry-pick)

saledjenic opened the pull request:
This is a re-implemented PR https://github.com/status-im/status-go/pull/6710 PR for the release.

saledjenic committed:
fix(wallet)_: add l1 fee to a base chain transactions

The Base chain is based on Optimism stack, so should follow the same fee calculation.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6712/1/) | 2025-06-25 14:17:58 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250625-141458-b84b92-pr6712.aar) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6712/1/) | 2025-06-25 14:18:04 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250625-141459-b84b92-pr6712.zip) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6712/1/) | 2025-06-25 14:19:50 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250625-141511-b84b92-pr6712.zip) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6712/1/) | 2025-06-25 14:20:47 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250625-141501-b84b92-pr6712.zip) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6712/1/) | 2025-06-25 14:21:00 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250625-141507-b84b92-pr6712.zip) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6712/1/) | 2025-06-25 14:21:00 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250625-141504-b84b92-pr6712.zip) |
| :heavy_check_mark: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6712/1/) | 2025-06-25 14:25:03 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6712/1//consoleText) |
| :heavy_multiplication_x: | b84b92cb | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/1/) | 2025-06-25 14:41:36 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/1//consoleText) |
| :heavy_multiplication_x: | b84b92cb | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/2/) | 2025-06-27 07:51:15 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/2//consoleText) |
| :heavy_multiplication_x: | b84b92cb | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/3/) | 2025-06-27 08:34:06 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6712/2/) | 2025-06-27 08:31:21 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250627-082818-498eed-pr6712.aar) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6712/2/) | 2025-06-27 08:31:24 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250627-082821-498eed-pr6712.zip) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6712/2/) | 2025-06-27 08:31:54 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250627-082832-498eed-pr6712.zip) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6712/2/) | 2025-06-27 08:31:59 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250627-082823-498eed-pr6712.zip) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6712/2/) | 2025-06-27 08:33:51 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250627-082821-498eed-pr6712.zip) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6712/2/) | 2025-06-27 08:34:18 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250627-082828-498eed-pr6712.zip) |
| :heavy_check_mark: | 498eedca | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6712/2/) | 2025-06-27 08:35:58 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6712/2//consoleText) |
| :heavy_check_mark: | 498eedca | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/4/) | 2025-06-27 09:01:29 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6712/4//consoleText) |

jrainville reviewed (approved):
Looks good. Is it possible to add a test?

saledjenic commented:
> Looks good. Is it possible to add a test?

The only valuable test would be a functional test for this, but it's not possible to make it (at least not at this stage, cause we have the Anvil chain we have mimics Ethereum mainnet, not Status chain). 

What can be added is a unit test, but no values in it, cause the only thing I could do there is to ensure that `NoBaseFee` and `NoPriorityFee` are set to false for the Status chain, but that's all artificial, no point in such a test.

saledjenic committed:
chore(wallet)_: fee handling improvements for Status Network Sepolia

- Remove unused functions `HasNoBaseFee` and `HasNoPriorityFee` from wallet utilities since that should not be hardcoded, but dynamically evaluated.
- A temporary patch was added to the `getEIP1559SuggestedFees` function to handle the specificities of the Status chain.
